internal/kibana/security_detection_rule/schema.go

```diff
@@ -913,3 +940,47 @@ func getThreatSubtechniqueElementType() attr.Type {
 	techniqueType := threatType.AttributeTypes()["technique"].(attr.TypeWithElementType).ElementType().(attr.TypeWithAttributeTypes)
 	return techniqueType.AttributeTypes()["subtechnique"].(attr.TypeWithElementType).ElementType()
 }
+
+// ValidateConfig validates the configuration for a security detection rule resource.
+// It ensures that the configuration meets the following requirements:
+//
+// - For rule types "esql" and "machine_learning", no additional validation is performed
+// - For other rule types, exactly one of 'index' or 'data_view_id' must be specified
+// - Both 'index' and 'data_view_id' cannot be set simultaneously
+//
+// The function adds appropriate error diagnostics if validation fails.
+func (r securityDetectionRuleResource) ValidateConfig(ctx context.Context, req resource.ValidateConfigRequest, resp *resource.ValidateConfigResponse) {
+	var data SecurityDetectionRuleData
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	t, diag := data.Type.ToStringValue(ctx)
+	resp.Diagnostics.Append(diag...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if t.ValueString() == "esql" || t.ValueString() == "machine_learning" {
```

Comment:
Isn't it already and StringValue?

```suggestion
	if data.Type..ValueString() == "esql" || data.Type..ValueString() == "machine_learning" {
```